plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -386,6 +393,36 @@ public Optional<String> getTableComment(ResultSet resultSet)
         return Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
     }
 
+    @Override
+    public Iterator<RelationColumnsMetadata> getAllTableColumns(ConnectorSession session, Optional<String> schema)
+    {
+        Collection<String> schemaNames;
+        if (schema.isEmpty()) {
+            schemaNames = getSchemaNames(session);
+        }
+        else {
+            schemaNames = ImmutableList.of(schema.get());
+        }
+
+        return schemaNames.stream()
+                .map(schemaName -> super.getAllTableColumns(session, Optional.of(schemaName)))
+                .flatMap(iterator -> StreamSupport.stream(
+                        Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false))
+                .iterator();
+    }
+
+    @Override
+    protected ResultSet getAllTableColumns(Connection connection, Optional<String> remoteSchemaName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getColumns(
+                metadata.getConnection().getCatalog(),
+                escapeObjectNameForMetadataQuery(remoteSchemaName, metadata.getSearchStringEscape()).orElse(null),
+                null,
+                null);
```

Comment:
Does the new logic get columns from synonyms when `oracle.synonyms.enabled` is enabled?